api/client/client.go

```diff
@@ -1976,21 +1977,29 @@ func (c *Client) GetTokens(ctx context.Context) ([]types.ProvisionToken, error)
 
 // UpsertToken creates or updates a provision token.
 func (c *Client) UpsertToken(ctx context.Context, token types.ProvisionToken) error {
-	tokenV2, ok := token.(*types.ProvisionTokenV2)
-	if !ok {
+	var err error
```

Comment:
This solution fails to account for scenarios where the client is /newer/ than the server - for example, if the proxy svc is newer than the auth service, it may attempt to submit a V3 token.

It feels like the appropriate solution is to offer a `.V2()` transformer on V3 tokens, and in the case that the V3 RPC fails with unimplemented, a V2 token can be submitted instead? Part of me thinks that deferring to the implementer makes sense here, in some situations you will not be able to convert a V3 token to a V2 token if it includes a newer token provider.

Reply:
If the client automatically converted from v3 to v2 in case of old server, we'd need to emit an error to indicate they've made use of a field which isn't compatible with V2.